### ISSUE TITLE
False Negative: Decorator signatures are unverified

### ISSUE THREAD

**Issue by deckar01** [PyCQA/pylint#1225]
### Steps to reproduce

```py
def decorator(func):
    def wrap():
        return func('test')
    return wrap

@decorator
def simple():
    return 'test'

simple()
```

### Current behavior

Pylint does not detect any errors.

### Expected behavior

Pylint should detect that `func` is called with too many arguments.

### pylint --version output

```
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 2.7.12 (default, Jun 28 2016, 06:57:42) [GCC]
```

### Related issues

#259

**Comment by PCManticore** [comment]
Closing as duplicate.

**Comment by deckar01** [comment]
@PCManticore I took a closer look at it, and this is not actually checked with function composition either, so it is less related to #259 than I though.

```py
def dec(func):
    def wrap():
        return func('test')
    return wrap

def simple():
    return 'test'

hard = dec(simple)

hard()
```

Boiling it down further illustrates the difference.

```py
def call(func):
    return func('test')

def simple():
    return 'test'

call(simple)
```

This code is guaranteed to error, but pylint doesn't report any errors.

**Comment by PCManticore** [comment]
This looks like an example of https://github.com/PyCQA/astroid/issues/177. This, combined with lack of flow control understanding across the inference, plus lack of understanding decorators, that stem from the first two points, results in the problems you are noticing. We know about it, it's not trivial to fix, but it's on our roadmap.
